apisix/core/config_etcd.lua

```diff
@@ -123,6 +124,11 @@ local function waitdir(etcd_cli, key, modified_index, timeout)
         res, err = res_func()
     end
 
+    if http_cli then
+        etcd_cli:watchcancel(http_cli)
```

Comment:
need test case.

we can use this way: write a log and confirm it in test case

Reply:
thx, test added